src/pages/physical-count-itemDetails/index.js

```diff
@@ -0,0 +1,633 @@
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useContext, useEffect, useState, useRef } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Button, Grid } from '@mui/material'
+import { ControlContext } from 'src/providers/ControlContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SCRepository } from 'src/repositories/SCRepository'
+import FormShell from 'src/components/Shared/FormShell'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { useForm } from 'src/hooks/form'
+import * as yup from 'yup'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { SystemChecks } from 'src/resources/SystemChecks'
+import toast from 'react-hot-toast'
+
+const PhysicalCountItemDe = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const [siteStore, setSiteStore] = useState([])
+  const [controllerStore, setControllerStore] = useState([])
+  const [filteredItems, setFilteredItems] = useState([])
```

Comment:
can we remove this state and add const directly in the form?
example: (not sure of the code)
const filteredItems=  formik.values.rows
        .filter(item => item.metalId && item.metalId.toString().trim() !== '')
        .map(item => ({
          qty: item.countedQty,
          metalRef: null,
          metalId: item.metalId,
          metalPurity: item.metalPurity,
          weight: item.weight,
          priceType: item.priceType
        }))

Reply:
this will affect the current functionality